packer/scripts/macos/macos-agentsetup.sh

```diff
@@ -13,29 +13,42 @@ sudo chown -R ec2-user:staff /var/jenkins
 /usr/local/bin/brew install grep
 /usr/local/bin/brew install wget 
 /usr/local/bin/brew install maven
+/usr/local/bin/brew install dpkg
 
-## Install MacPorts, jEnv, setup java8,11,17,21 and python, then set default java to 11
+## Array of JDK versions in form of version<space>path<space>priority
+jdk_versions=(
+    "8 /jdk8u392-b08/OpenJDK8U-jdk_x64_mac_hotspot_8u392b08.tar.gz 1"
+    "11 /jdk-11.0.21%2B9/OpenJDK11U-jdk_x64_mac_hotspot_11.0.21_9.tar.gz 100"
+    "17 /jdk-17.0.9%2B9/OpenJDK17U-jdk_x64_mac_hotspot_17.0.9_9.tar.gz 1"
+    "19 /jdk-19.0.2%2B7/OpenJDK19U-jdk_x64_mac_hotspot_19.0.2_7.tar.gz 1"
+    "20 /jdk-20.0.2%2B9/OpenJDK20U-jdk_x64_mac_hotspot_20.0.2_9.tar.gz 1"
+    "21 /jdk-21.0.1%2B12/OpenJDK21U-jdk_x64_mac_hotspot_21.0.1_12.tar.gz 1"
+)
+
+## Loop through JDK versions and install them
+for version_info in "${jdk_versions[@]}"; do
+    version_num=$(echo "$version_info" | cut -d ' ' -f 1)
+    version_path=$(echo "$version_info" | cut -d ' ' -f 2)
+    version_priority=$(echo "$version_info" | cut -d ' ' -f 3)
+    url="https://github.com/adoptium/temurin${version_num}-binaries/releases/download${version_path}"
+    sudo mkdir -p "/opt/java/openjdk-${version_num}/"
+    /usr/local/bin/wget "$url" -O "OpenJDK${version_num}-jdk_x64_mac_hotspot_${version_num}.tar.gz"
```

Comment:
Since you are already using the `-O` you can make the file name simplier such as `openjdk-11.tar.gz`.

Reply:
^^ right .. was trying to preserve the original name for no special reasons .. will be easier to do the way as you mentioned ,, let me get that update too , thanks